Inc/ST-LIB_HIGH/Protections/Protection.hpp

```diff
@@ -4,24 +4,20 @@
 #include "ErrorHandler/ErrorHandler.hpp"
 #include "Boundary.hpp"
 
-namespace Protections{
-enum FaultType : uint8_t{
-	FAULT = 0,
-	WARNING,
-    OK
-};
-}
+
 
 class Protection{
 private:
     char* name = nullptr;
-    vector<unique_ptr<BoundaryInterface>> boundaries;
-    BoundaryInterface* jumped_protection = nullptr;
+    vector<shared_ptr<BoundaryInterface>> boundaries;
+    BoundaryInterface* fault_protection = nullptr;
     static constexpr Protections::FaultType fault_type = Protections::FaultType::FAULT;
+    uint8_t triggered_protecions_idx[4]{};
 public:
+    vector<shared_ptr<BoundaryInterface>> warnings_triggered;
     template<class Type, ProtectionType... Protector, template<class,ProtectionType> class Boundaries>
     Protection(Type* src, Boundaries<Type,Protector>... protectors) {
-        (boundaries.push_back(unique_ptr<BoundaryInterface>(new Boundary<Type,Protector>(src,protectors))), ...);
+        (boundaries.push_back(shared_ptr<BoundaryInterface>(new Boundary<Type,Protector>(src,protectors))), ...);
```

Comment:
why do you want the reference count?

Reply:
because unique_ptr is a pain in the *, and it cannot be copied, so in order to maintain the RAII principle, I used a shared_ptr and after sending the warning I delete the one I copied.

Reply:
that's why I added my own comment with the godbolt link, https://godbolt.org/z/dd1aofzvP